tests/driver_n25q128/main.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test for the N25Q128 serial flash memory
+ *
+ * @author      René Herthel <rene-herthel@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "xtimer.h"
+#include "board.h"
+#include "periph/pm.h"
+#include "shell.h"
+#include "n25q128.h"
+
+/*
+ * @brief   The size of the generic buffer.
+ */
+#define BUF_SIZE     (20U)
+
+/*
+ * @brief   The device, used to get access to the n25q128 serial flash memory.
+ */
+static n25q128_dev_t n25q128;
+
+/*
+ * @brief   Generic buffer used for sending and receiving.
+ */
+static uint8_t buf[BUF_SIZE];
+
+static inline void _print_bytes(char * title, uint8_t * data, size_t len)
+{
+    printf("\n%4s; len = %d\n\n", title, len);
+    for (size_t i = 0; i < len; i++) {
+        printf("  %2i ", (int)i);
+    }
+    printf("\n");
+    printf("\n ");
+    for (size_t i = 0; i < len; i++) {
+        printf(" 0x%02x", (int)data[i]);
+    }
+    printf("\n ");
+    for (size_t i = 0; i < len; i++) {
+        if (data[i] < ' ' || data[i] > '~') {
+            printf("  ?? ");
+        }
+        else {
+            printf("   %c ", (char)data[i]);
+        }
+    }
+    printf("\n\n");
+}
+
+static int cmd_init(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+    int status = 0;
+
+    puts("cmd_init: Initialize the N25Q128 flash memory.\n");
+
+    // XXX: Specific configuration for iotlab-m3 nodes.
+    n25q128.conf.bus = EXTFLASH_SPI;
```

Comment:
Configuration should be done as described in the [drivers guide](https://doc.riot-os.org/driver-guide.html#driver-guide-config)